src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -15,18 +15,30 @@ Implements the extended Hubbard model on a one-dimensional chain in real space.
 * `t`: the hopping strength
 
 """
-struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T} <: AbstractHamiltonian{TT}
+struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T,PITWISTED,HARDWALL} <: AbstractHamiltonian{TT}
     add::A
 end
 
 # addr for compatibility.
-function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0)
-    U, V, T = promote(float(u), float(v), float(t))
-    return ExtendedHubbardReal1D{typeof(U),typeof(addr),U,V,T}(addr)
+function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0, pitwisted::Bool = false, hardwall::Bool = false)
+    if (pitwisted,hardwall)==(true,true)
+        println("Hardwall and pitwisted boundaries cannot be called together")
```

Comment:
You want this to throw an error:
```suggestion
        throw(ArgumentError("Hardwall and pitwisted boundaries cannot be called together"))
```